ocaml/xapi/repository.ml

```diff
@@ -375,57 +375,64 @@ let create_pool_repository ~__context ~self =
   let repo_dir = Filename.concat !Xapi_globs.local_pool_repo_dir repo_name in
   match Sys.file_exists repo_dir with
   | true -> (
-    try
-      let cachedir = get_repo_config repo_name "cachedir" in
-      let group_params =
-        match
-          RepoMetaData.(
-            of_xml_file (Filename.concat cachedir "repomd.xml") Group
-          )
+      let repo_md_dir = Filename.concat repo_dir "repodata" in
```

Comment:
Is it also easy to check if the "reposync" has synced the complete metadata in `Repository.sync` function?
If yes, the change would be clear in:
```
  let sync_repos ~__context ~self ~repos ~force ~token ~token_id =
    let open Repository in
    repos
    |> List.iter (fun repo ->
           if force then cleanup_pool_repo ~__context ~self:repo ;
           let complete = sync ~__context ~self:repo ~token ~token_id in
           (* Dnf sync all the metadata including updateinfo,
            * Thus no need to re-create pool repository *)
           if Pkgs.manager = Yum and complete = false then
             create_pool_repository ~__context ~self:repo
       ) ;
```

Reply:
In this way, we need to update the process to distinguish the new reposync method as "complete" and the old method as "not complete," which seems even more confusing.

Reply:
I think **full_complete** might be a more accurate description. I've moved the logic for checking whether "reposync" has synced the complete metadata to `Repository.sync`

Reply:
Language: it would be "fully complete" but I don't like that either. Something is either complete or not.